[upnpcontrol] Error when parsing URI

I am on 4.2 S4020 (the error has been present since I started with OH 3.0). The bridge connects correctly to the device at startup but eventually goes offline. There is a Java error in the console that seems to be related to the UPNP service. Error and trace output below.

https://community.openhab.org/t/upnp-device-not-registered-error/155081

```
2024-03-31 07:39:04.828 [TRACE] [internal.handler.UpnpRendererHandler] - UPnP device Arylic Kitchen received variable CurrentTrackMetaData with value <?xml version="1.0" encoding="UTF-8"?> <DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:song="www.wiimu.com/song/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"> <upnp:class>object.item.audioItem.musicTrack</upnp:class> <item id="0"> <song:subid></song:subid> <song:description></song:description> <song:skiplimit>0</song:skiplimit> <song:id></song:id> <song:like>0</song:like> <song:singerid>0</song:singerid> <song:albumid>0</song:albumid> <song:rate_hz></song:rate_hz> <song:format_s></song:format_s> <song:bitrate>0</song:bitrate> <res protocolInfo="http-get:*:audio/mpeg:DLNA.ORG_PN=MP3;DLNA.ORG_OP=01;" duration="00:00:00.000"></res> <dc:title></dc:title> <dc:creator></dc:creator> <upnp:artist></upnp:artist> <upnp:album></upnp:album> <upnp:albumArtURI>un_known</upnp:albumArtURI> </item> </DIDL-Lite>  from service AVTransport
2024-03-31 07:39:04.835 [TRACE] [internal.handler.UpnpRendererHandler] - Renderer Arylic Kitchen, received media ID: 0
2024-03-31 07:39:04.836 [TRACE] [internal.handler.UpnpRendererHandler] - Current queue res: 
2024-03-31 07:39:04.836 [TRACE] [internal.handler.UpnpRendererHandler] - Updated media res: 
2024-03-31 07:39:04.837 [TRACE] [internal.handler.UpnpRendererHandler] - Received meta data is for current entry: true
2024-03-31 07:39:04.838 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.IllegalArgumentException: Illegal character in path at index 8: un_known 
	at java.net.URI.create(URI.java:906) ~[?:?]
	at org.eclipse.jetty.client.HttpClient.newRequest(HttpClient.java:473) ~[?:?]
	at org.openhab.core.io.net.http.HttpUtil.executeUrlAndGetReponse(HttpUtil.java:212) ~[?:?]
	at org.openhab.core.io.net.http.HttpUtil.downloadData(HttpUtil.java:443) ~[?:?]
	at org.openhab.core.io.net.http.HttpUtil.downloadImage(HttpUtil.java:406) ~[?:?]
	at org.openhab.core.io.net.http.HttpUtil.downloadImage(HttpUtil.java:377) ~[?:?]
	at org.openhab.core.io.net.http.HttpUtil.downloadImage(HttpUtil.java:363) ~[?:?]
	at org.openhab.binding.upnpcontrol.internal.handler.UpnpRendererHandler.updateMetaDataState(UpnpRendererHandler.java:1654) ~[?:?]
	at org.openhab.binding.upnpcontrol.internal.handler.UpnpRendererHandler.onValueReceivedCurrentMetaData(UpnpRendererHandler.java:1306) ~[?:?]
	at org.openhab.binding.upnpcontrol.internal.handler.UpnpRendererHandler.onValueReceived(UpnpRendererHandler.java:1089) ~[?:?]
	at org.openhab.binding.upnpcontrol.internal.handler.UpnpRendererHandler.lambda$3(UpnpRendererHandler.java:1161) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.net.URISyntaxException: Illegal character in path at index 8: un_known 
	at java.net.URI$Parser.fail(URI.java:2976) ~[?:?]
	at java.net.URI$Parser.checkChars(URI.java:3147) ~[?:?]
	at java.net.URI$Parser.parseHierarchical(URI.java:3229) ~[?:?]
	at java.net.URI$Parser.parse(URI.java:3188) ~[?:?]
	at java.net.URI.<init>(URI.java:623) ~[?:?]
	at java.net.URI.create(URI.java:904) ~[?:?]
	... 16 more
```

--- a/bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/handler/UpnpRendererHandler.java
+++ b/bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/handler/UpnpRendererHandler.java
@@ -1651,7 +1651,12 @@ public class UpnpRendererHandler extends UpnpHandler {
             if (media.getAlbumArtUri().isEmpty() || media.getAlbumArtUri().contains("DefaultAlbumCover")) {
                 updateState(ALBUM_ART, UnDefType.UNDEF);
             } else {
-                State albumArt = HttpUtil.downloadImage(media.getAlbumArtUri());
+                State albumArt = null;
+                try {
+                    albumArt = HttpUtil.downloadImage(media.getAlbumArtUri());
+                } catch (IllegalArgumentException e) {
+                    logger.debug("Invalid album art URI: {}", media.getAlbumArtUri(), e);
+                }
                 if (albumArt == null) {
                     logger.debug("Failed to download the content of album art from URL {}", media.getAlbumArtUri());
                     if (!isCurrent) {

